src/ex2/script.js

```diff
@@ -0,0 +1,146 @@
+        addBtn.disabled = true;
+        this.updateForm();
+    }
+
+    // function to remove from DOM task that have been clicked
+    removeLiTask(li) {
+        taskList.removeChild(li);
+        this.updateForm();
+    }
+
+    //update the button and the placeHolder if necessary
+    updateForm() {
+        pedingTaskText.innerHTML = "You have " + itemManager.taskArr.length + " pending tasks"
+        if (itemManager.taskArr.length === 0) { //if there are no task any more disable the clear and sort button
+            clearAllBtn.disabled = true;
+            sortAllBtn.disabled = true;
+            placeHolder.classList.remove("hidden");
+        }
+        else {
+            clearAllBtn.disabled = false;
+            sortAllBtn.disabled = false;
+            placeHolder.classList.add("hidden");
+        }
+    }
+
+    //function to insert the all tasks from the localStorage to the html list element
+    renderAllTasks() {
+        itemManager.taskArr.forEach((taskValue) => {
+            taskList.appendChild(this.createDOMTask(taskValue));
+        });
+        this.updateForm();
+    }
+
+    //Function to create the li element and its childs and assign a click event to them
+    createDOMTask(taskValue) {
+        const li = document.createElement("li");
+        const span = document.createElement("span");
+        const i = document.createElement("i");
+        li.innerHTML = taskValue;
+        i.classList.add("fa", "fa-trash-o");
+        span.appendChild(i);
+        li.appendChild(span);
+        span.onclick = (e) => {
+            e.stopPropagation(); //stop the parent elemnt to listen to click event
+            itemManager.removeTask(taskValue);
+            this.removeLiTask(li);
+        };
+        li.onclick = () => alert(taskValue);
+        li.classList.add('new-li');
+        return li;
+    }
+
+    addTodoAndRender() {
+        //input a number
+        if (/^\d*$/.test(taskInput.value)) {
+            pokemonClient.getPokemon(taskInput.value).then((pokemon) => {
+                if (pokemon) {
+                    itemManager.addNewTask(`Catch ${pokemon}`);
+                    this.addLiTask(`Catch ${pokemon}`);
+                } else {
+                    itemManager.addNewTask(`Pokemon with ${taskInput.value} was not found`);
+                    this.addLiTask(`Pokemon with ${taskInput.value} was not found`);
+                }
+            })
+        }
+        //inputs a comma separated list of IDs 
+        else if (/^\d+(\,\d+)+$/.test(taskInput.value)) {
+            pokemonClient.getPokemons(taskInput.value).then((pokemons) => {
+                if (pokemons) {
+                    pokemons.forEach((pokemon) => {
+                        itemManager.addNewTask(`Catch ${pokemon}`);
+                        this.addLiTask(`Catch ${pokemon}`);
+                    })
+                } else {
+                    itemManager.addNewTask(`faild to fetch pokemon with this input: ${taskInput.value}`);
```

Comment:
there is a duplication here of building the text, also it's not a good practice that you have this separation and that there's not one function that does both things